packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -289,6 +289,15 @@ mixin WidgetsBinding on BindingBase, SchedulerBinding, GestureBinding, RendererB
           return _forceRebuild();
         },
       );
+
+      registerServiceExtension(
```

Comment:
Add tests verifying the service extension returns false before a frame and true after. There is also a generic test that counts the # of service extensions that should now be failing.  There are existing tests that show intercepting calls to register service extensions so they are painless to test.

Reply:
signalServiceExtension returns no value, so I believe I'll still have to use registerServiceExtension.

Reply:
you are right. cases for signalServiceExtension dump to the console instead of returning.